code/modules/admin/verbs/atmosdebug.dm

```diff
@@ -1,59 +1,85 @@
-/client/proc/check_plumbing()
+#define ANNOTATE_OBJECT(object) testing ? "[get_area(object)] (estimated location: [english_list(object.check_shuttle_offset()) || "[object.x], [object.y]"])" : ADMIN_VERBOSEJMP(object)
+
+/atom/proc/check_shuttle_offset()
+	if(!SSshuttle.initialized)
+		return
+	var/obj/docking_port/mobile/shuttle = SSshuttle.get_containing_shuttle(src)
+	if(!shuttle)
+		return
+	var/list/bounds = shuttle.return_coords()
+	var/shuttle_dir = turn(shuttle.port_direction, -dir2angle(shuttle.dir)) // evil code
+	switch(shuttle_dir)
+		if(SOUTH)
+			return list(x + 1 - min(bounds[1], bounds[3]), y + 1 - min(bounds[2], bounds[4]))
+		if(NORTH)
+			return list(-(x - 1 - max(bounds[1], bounds[3])), y + 1 - min(bounds[2], bounds[4]))
+		if(EAST)
+			return list(y + 1 - min(bounds[2], bounds[4]), x + 1 - min(bounds[1], bounds[3]))
+		if(WEST)
+			return list(y + 1 - min(bounds[2], bounds[4]), -(x - 1 - max(bounds[1], bounds[3])))
```

Comment:
I'm going to be blunt, this *works* but I really need help to make this actually work correctly 100% of the time. If anyone is willing and able to help, please reach out to me, the math is very much off and the macro is just not how it should be.